desktop/apps/editorial_features/components/venice_2017/client/index.coffee

```diff
@@ -118,25 +119,25 @@ module.exports = class VeniceView extends Backbone.View
         new FlashMessage message: 'Whoops, there was an error. Please try again.'
         @$(e.currentTarget).removeClass 'is-loading'
       success: (res) =>
+        analyticsHooks.trigger 'email:signup', {user_email: @email}
         new FlashMessage message: 'Thank you for signing up.'
         @$(e.currentTarget).removeClass 'is-loading'
-        @$(e.currentTarget).prev('input').val('')
-        @$(e.currentTarget).closest('.venice-overlay__subscribe-form').fadeOut()
-        @$(e.currentTarget).closest('.venice-overlay__subscribe').find('.venice-overlay__cta-button').fadeIn()
+        @$('.venice-overlay__cta').hide()
 
   setupFollowButtons: ->
     @artists = []
     @$('.artist-follow').each (i, artist) =>
-      @artists.push id: $(artist).data('id')
+      @artists.push {id: $(artist).data('id'), _id: $(artist).attr('artist-id')}
     @followButtons = @artists.map (artist) =>
-      artist = new Artist id: artist.id
+      artist = new Artist id: artist.id, _id: artist._id
       new FollowButton
         el: @$(".artist-follow[data-id='#{artist.id}']")
         following: @following
         modelName: 'artist'
         model: artist
-        context_page: "Article page" # TODO: add venice specific tracking
+        context_page: 'venice_biennale_2017'
         context_module: 'article_artist_follow'
+        entity_id: artist.id
```

Comment:
For the jitter you were talking about, maybe it has to do with calling [this](https://github.com/artsy/force/pull/1327/files#diff-2b8b6d65a72cb42cf276bc67340b1d07R56) on every carousel turn.

This `setupFollowButtons` function is selecting all follow buttons (`@$('.artist-follow').each`) which means there's probably an unnecessary amount of follow buttons being initialized/placed on the page?